src/app/register/_components/EmailPasswordBox.tsx

```diff
@@ -0,0 +1,119 @@
+"use client";
+
+import * as React from "react";
+import { Card, CardContent } from "@/components/ui/card";
+import { Input } from "@/components/ui/input";
+import { Label } from "@/components/ui/label";
+
+interface EmailPasswordBoxProps {
+  setPasswordMatch: (isMatch: boolean) => void;
+  setPasswordValid: (isValid: boolean) => void;
+  setEmailValid: (isValid: boolean) => void;
+  setEmailAlreadyReg: (isValid: boolean) => void;
+}
+
+export default function EmailPasswordBox(props: EmailPasswordBoxProps) {
+  const checkConfirmPassword = (props: EmailPasswordBoxProps) => {
+    const password = document.getElementById("Password") as HTMLInputElement;
+    const confirmPassword = document.getElementById(
+      "Confirm Password",
+    ) as HTMLInputElement;
+    if (!!confirmPassword && !!password && confirmPassword.value != "") {
+      if (password.value == confirmPassword.value) {
+        (document.getElementById("message") as HTMLInputElement).style.color =
+          "green";
+        (document.getElementById("message") as HTMLInputElement).innerHTML = "";
+        props.setPasswordMatch(true);
+      } else {
+        (document.getElementById("message") as HTMLInputElement).style.color =
+          "red";
+        (document.getElementById("message") as HTMLInputElement).innerHTML =
+          "not matching";
+        props.setPasswordMatch(false);
+      }
+    }
+  };
+
+  const checkValidPassword = (props: EmailPasswordBoxProps) => {
+    const password = document.getElementById("Password") as HTMLInputElement;
+    if (!!password) {
+      if (password.value.length >= 8) {
+        // add additional condition here
+        props.setPasswordValid(true);
+      } else {
+        props.setPasswordValid(false);
+      }
+    }
+  };
+
+  const checkValidEmail = (props: EmailPasswordBoxProps) => {
+    const email = document.getElementById("Email") as HTMLInputElement;
+    if (!!email) {
+      const validRegex =
+        /^[a-zA-Z0-9.!#$%&'*+/=?^_`{|}~-]+@[a-zA-Z0-9-]+(?:\.[a-zA-Z0-9-]+)*$/;
+      console.log(
+        `check email: ${email.value} \n valid = `,
+        email.value.match(validRegex),
+      ); //DEBUGGING
+      if (email.value.match(validRegex)) {
+        props.setEmailValid(true);
+      } else {
+        props.setEmailValid(false);
+      }
+
+      if (true) {
+        // add condition here
+        props.setEmailAlreadyReg(true);
+      } else {
+        props.setEmailAlreadyReg(false);
+      }
+    }
+  };
```

Comment:
don't pass props into function like this, you already declare function inside component so you can access it directly anyway.